spring-integration-core/src/test/java/org/springframework/integration/endpoint/PseudoTransactionalMessageSourceTests.java

```diff
@@ -342,18 +342,25 @@ public void testRollbackWithManagerUsingStatus() {
 
 	@Test
 	public void testInt2777UnboundResourceAfterTransactionComplete() {
+		MessageSource<?> source = () -> null;
 		SourcePollingChannelAdapter adapter = new SourcePollingChannelAdapter();
+		adapter.setSource(source);
+		adapter.setOutputChannelName("bar");
+		adapter.setBeanFactory(new TestUtils.TestApplicationContext());
+		adapter.onInit();
 
-		adapter.setSource(() -> null);
-
+		TransactionSynchronizationManager.initSynchronization();
 		TransactionSynchronizationManager.setActualTransactionActive(true);
 		doPoll(adapter);
 		TransactionSynchronizationManager.setActualTransactionActive(false);
+		TransactionSynchronizationManager.unbindResource(source);
 
 		// Before INT-2777 this test was failed here
 		TransactionSynchronizationManager.setActualTransactionActive(true);
 		doPoll(adapter);
 		TransactionSynchronizationManager.setActualTransactionActive(false);
+		TransactionSynchronizationManager.clearSynchronization();
+		TransactionSynchronizationManager.unbindResource(source);
```

Comment:
This test did not bind resources to the `TransactionSynchronizationManager` before my changes because the `TransactionSynchronizationFactory` was null. Now that it does bind resources they must be cleared otherwise subsequent tests fail.